testkit/src/builder.rs

```diff
@@ -156,6 +162,17 @@ impl TestKitBuilder {
         self
     }
 
+    /// Adds a runtime factory to the testkit.
+    pub fn with_runtime_factory(
+        mut self,
+        runtime_factory: Box<dyn RuntimeFactory>,
+        instances: impl IntoIterator<Item = InstanceConfig>,
```

Comment:
Why we pass instances to method `with_runtime_factory`? Can we modify `with_service` method instead?

Reply:
Well, they were different methods initially, then I decided to combine them because every runtime has its own set of instances and this approach prevents a user from forgetting passing instances in addition to runtime factory.

Reply:
How the user can forget to pass services, if this is the main feature of TestKit - to create a blockchain with specified services list?